dask/tests/test_order.py

```diff
@@ -272,27 +465,25 @@ def test_type_comparisions_ok(abcde):
     order(dsk)  # this doesn't err
 
 
-def test_prefer_short_dependents(abcde):
+def test_favor_longest_critical_path(abcde):
     r"""
 
-         a
-         |
-      d  b  e
-       \ | /
-         c
+       a
+       |
+    d  b  e
+     \ | /
+       c
```

Comment:
I believe the argument of the earlier version is wrong. Writing again out what the pressure is over time, yields

Single threaded

This PR

| Step     | 0 | 1 | 2 | 3 | 4 |
| -        | - | -- | -- | -- |-- |
| T1       | c | b  | a | e | d |
| Pressure | 1 | 2  | 2  | 3  | 3 |

Main

| Step     | 0 | 1 | 2 | 3 | 4 |
| -        | - | -- | -- | -- |-- |
| T1       | c | d  | e | b | a |
| Pressure | 1 | 2  | 3  | 3  | 3 |


Multi threaded

This PR

| Step     | 0 | 1 | 2 |
| -        | - | -- | -- |
| T1       | c | b  | a |
| T2       | x | d | e |
| Pressure | 1 | 3  | 3 |


Main

| Step     | 0 | 1 | 2 | 3 |
| -        | - | -- | -- |-- |
| T1       | c | d | b | a |
| T2       | x | e | x | x |
| Pressure | 1 | 3 | 3 | 3 |



Of course, this assumes that all tasks weigh the same. If we assumed that source tasks weight more, this is different but this is not universally true (often but not always).

Favoring the longest critical path, however, benefits execution times in multi threaded environments (fewer steps, again assuming equal weight / runtime)